DevCycleTests/Models/DVCVariableTests.swift

```diff
@@ -95,12 +95,84 @@ class DVCVariableTests: XCTestCase {
         XCTAssertNotNil(variable)
         XCTAssertEqual(variable.key, "my_key")
         XCTAssertEqual(variable.value, 2.3)
-        XCTAssertEqual(variable.type, "Number")
+        XCTAssertEqual(variable.type, DVCVariableTypes.Number)
         XCTAssertEqual(variable.defaultValue, 1.1)
         XCTAssertFalse(variable.isDefaulted)
         XCTAssertNil(variable.evalReason)
     }
     
+    func testNumberDoubleVariableCreatedFromVariable() throws {
+        let data = """
+        {
+            "_id": "variable_id",
+            "key": "my_key",
+            "type": "Number",
+            "value": 2.3
+        }
+        """.data(using: .utf8)!
+        let variableDict = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed) as! [String: Any]
+        let variableFromApi = try Variable(from: variableDict)
+        let defaultDouble: Double = 1.1
+        let variable = DVCVariable(from: variableFromApi, defaultValue: defaultDouble)
+        XCTAssertEqual(variable.value, 2.3)
+        XCTAssertEqual(variable.type, DVCVariableTypes.Number)
+        XCTAssertEqual(variable.defaultValue, defaultDouble)
+    }
+
+    func testNumberNSNumberVariableCreatedFromVariable() throws {
+        let data = """
+        {
+            "_id": "variable_id",
+            "key": "my_key",
+            "type": "Number",
+            "value": 2.3
+        }
+        """.data(using: .utf8)!
+        let variableDict = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed) as! [String: Any]
+        let variableFromApi = try Variable(from: variableDict)
+        let defaultInt: NSNumber = 1
+        let variable = DVCVariable(from: variableFromApi, defaultValue: defaultInt)
+        XCTAssertEqual(variable.value, 2.3)
```

Comment:
what is the actual datatype of variable.value here?

Reply:
It is a NSNumber Double: 
<img width="935" alt="image" src="https://github.com/DevCycleHQ/ios-client-sdk/assets/1219069/d87065ed-f720-4f29-bb05-d8d591bd4b36">

Reply:
and isnt the defaultValue an int? Are those two types compatible with each other? Like what would the type of the value be if it defaulted instead

Reply:
yea `NSNumber` is basically a common bridge between all the lower level number types: https://developer.apple.com/documentation/foundation/nsnumber